### ISSUE TITLE
Rule blocked: CORERULES-9309 (FB3001): issue with is_inconsistent_across_datasets operator

### ISSUE THREAD

**Issue by marisawyckmans** [cdisc-org/cdisc-rules-engine#1223]
**Links to related JIRA Tickets**
[- https://jira.cdisc.org/projects/CORERULES/issues/CORERULES-9309](https://jira.cdisc.org/browse/CORERULES-9309?filter=-4&jql=project%20%3D%20CORERULES%20AND%20component%20in%20(%22FDA%20SDTMIG%20v3.2%22%2C%20%22FDA%20SDTMIG%20v3.3%22%2C%20%22FDA%20SDTMIG%20v3.4%22)%20AND%20text%20~%20%22FB3001%22%20ORDER%20BY%20summary%20ASC%2C%20created%20DESC)

**Rule Information**
- **Standard**: FDA (SDTM and SEND)
- **Rule ID**: FB3001
- **Rule Description**: Standard units should be consistent within the same assessment
  (having the same --TESTCD, --CAT, --SCAT, --SPEC, --METHOD values)

**Describe the bug**
This rule was programmed using the is_inconsistent_across_dataset operator. We have noticed that it gives unnecessary output for not done records (where --STRESU is empty). Even if all populated --STRESU are consistent for a given test combination, it sees the empty --STRESU of the not done records as a different / inconsistent value.
We tried adding a condition to the rule to only take into account non empty --STRESU, but it still gives output for inconsistent values, it just doesn't show the empty --STRESU in the output.

**Expected behavior**
It should be possible to exclude records with empty --STRESU so the is_inconsistent_across_dataset operator only takes into account the populated --STRESU.

**Screenshots**
Updated rule logic for empty STRESU
Check:
  all:
    - name: --STRESU
      operator: non_empty
    - name: --STRESU
      operator: is_inconsistent_across_dataset
      value:
        - --TESTCD
        - --CAT
        - --SCAT
        - --SPEC
        - --METHOD

Test data is available here: https://cdisc.sharepoint.com/sites/CORERules/Shared%20Documents/Forms/AllItems.aspx?id=%2Fsites%2FCORERules%2FShared%20Documents%2FunitTesting%2FFDA%20Business%20Rules%2FFB3001&viewid=ecd009a9%2Db73b%2D415b%2D9e41%2Df9d04a69edf4

Request and result for positive test data: 

[Rule - 2025-06-27T120136.266.zip](https://github.com/user-attachments/files/20944466/Rule.-.2025-06-27T120136.266.zip)

**Comment by gerrycampion** [comment]
@marisawyckmans 
It makes sense that you are seeing an issue even though you are specifying
```yaml 
- name: --STRESU
operator: non_empty
```
This is because it is flagging the records where --STRESU is populated like 'm' and 'g/L' because it is comparing it against other records that have the same --TESTCD, --CAT, --SCAT, --SPEC, --METHOD but have null --STRESU

There are possible solutions:
1. Add --STAT to the list of variables and remove --STRESU non_empty. This is just a rule change.
2. We could add something like a `filter:` parameter to the is_inconsistent_across_dataset operation to ignore the empties. This would require an update to the engine.

I think 1 is preferable because it is less complex. Would this work for you?

**Comment by marisawyckmans** [comment]
@gerrycampion 
we have updated the rule per your suggestion in option 1 and this works fine. Rule is published again. Many thanks for your help!

**Comment by SFJohnson24** [comment]
I am closing this issue as it seems it is resolved above, if I am incorrect--we can reopen
